Add files via upload

lonamle opened the pull request:
<!-- Example pull request description -->

# Purpose
Wheel speed IR sensor data collection

This file will standardize information needed for a pull request

## Implementation
Uses interrupt to get data from IR sensor. Also contains basic functions to test that sensor is working.

This file uses markdown to format the text into headers

## Testing 

Run code using PlatformIO. Need to figure out how to test the sensor properly.

View on github

## References
https://dronebotworkshop.com/robot-car-with-speed-sensors/ 

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#referencing-issues-and-pull-requests

## PR Review Rules 

1. Reviewer is responsible for resolving any threads they started
2. One commit per thread and if it's a widespread change, link the commit in the reply pls :)
3. ...

lonamle committed:
Add files via upload

added files for main.cpp and sensor driver code

lonamle committed:
Add files via upload

lonamle committed:
Add files via upload

using interrupt to get wheel speed data from IR sensor

rafguevara14 commented on main.cpp:
```diff
@@ -0,0 +1,87 @@
+
+#define tempConst 5 // adjustable temperature proportionality constant for testing
+
+uint16_t wheelPulses=120; // set at random number for testing
```
this could be a potential and maybe the potential issue! When dealing with variables and interrupts, you should declare variables as volatile.

The compiler may try to be a smart aleck and see that the ISR function never gets explicitly called. It may just assume wheelPulses never gets incremented and so will lazily never update the value of wheelPulses properly in the main code (even if wheelPulses is actually being incremented!). 

The ISR depends on hardware signals that the compiler wouldn't be aware of. By making it volatile, you're basically saying in plain english "I know more about what will happen to this variable than you. **Please don't do any smart optimizations**. Read the variable from memory every single time"

lonamle committed:
Add files via upload

working code for 1 sensor, duplicated to work with 4 sensors but not yet tested

lonamle committed:
Add files via upload

working code for 4 sensors

lonamle committed:
Add files via upload

timer library

Commit:
updated RPM calculation to format as float

Commit:
updated to calculate and output velocity value
